kernel/src/table_features/mod.rs

```diff
@@ -124,6 +150,424 @@ pub(crate) enum FeatureType {
+    enablement_check: EnablementCheck::EnabledIf(|props| props.append_only == Some(true)),
+};
+
+#[allow(dead_code)]
+static INVARIANTS_INFO: FeatureInfo = FeatureInfo {
+    name: "invariants",
+    min_reader_version: 1,
+    min_writer_version: 2,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static CHECK_CONSTRAINTS_INFO: FeatureInfo = FeatureInfo {
+    name: "checkConstraints",
+    min_reader_version: 1,
+    min_writer_version: 3,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static CHANGE_DATA_FEED_INFO: FeatureInfo = FeatureInfo {
+    name: "changeDataFeed",
+    min_reader_version: 1,
+    min_writer_version: 4,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_change_data_feed == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static GENERATED_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "generatedColumns",
+    min_reader_version: 1,
+    min_writer_version: 4,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IDENTITY_COLUMNS_INFO: FeatureInfo = FeatureInfo {
+    name: "identityColumns",
+    min_reader_version: 1,
+    min_writer_version: 6,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::NotSupported,
+    enablement_check: EnablementCheck::AlwaysIfSupported,
+};
+
+#[allow(dead_code)]
+static IN_COMMIT_TIMESTAMP_INFO: FeatureInfo = FeatureInfo {
+    name: "inCommitTimestamp",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[],
+    read_support: KernelSupport::NotSupported,
+    write_support: KernelSupport::Supported,
+    enablement_check: EnablementCheck::EnabledIf(|props| {
+        props.enable_in_commit_timestamps == Some(true)
+    }),
+};
+
+#[allow(dead_code)]
+static ROW_TRACKING_INFO: FeatureInfo = FeatureInfo {
+    name: "rowTracking",
+    min_reader_version: 3,
+    min_writer_version: 7,
+    feature_type: FeatureType::Writer,
+    feature_requirements: &[FeatureRequirement::Supported(TableFeature::DomainMetadata)],
+    read_support: KernelSupport::NotSupported,
```

Comment:
We support reading row tracking.


```suggestion
    read_support: KernelSupport::Supported,
```